src/ec2/model.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { Session } from 'aws-sdk/clients/ssm'
+import { Ec2Selection } from './utils'
+import { getOrInstallCli } from '../shared/utilities/cliUtils'
+import { isCloud9 } from '../shared/extensionUtilities'
+import { withoutShellIntegration } from '../ecs/commands'
```

Comment:
Followup: does it make sense to extract most of `ecs/util.ts` and `ecs/commands.ts` into something like `shared/remoteSession.ts` ... most of it looks re-usable. Just need to parameterize `client` and telemetry in a few places.

Reply:
That's a good point. I was using that code as a reference. The main difference is how we start and work with a session. However, splitting this up into smaller units should yield some shared components.